substrate/client/cli/src/config.rs

```diff
@@ -185,7 +185,17 @@ pub trait CliConfiguration<DCV: DefaultConfigurationValues = ()>: Sized {
 			)
 		} else {
 			NetworkConfiguration::new(node_name, client_id, node_key, Some(net_config_dir))
-		})
+		};
+
+		if is_validator && network_config.public_addresses.is_empty() {
+			eprintln!(
```

Comment:
It would be better to use `log::warn!` for this, to properly handle all the logging backends

Reply:
Yep, although the logger is not initialized yet :(

We are fetching the config first, then initializing the logger with that config:

https://github.com/paritytech/polkadot-sdk/blob/dc5f96dc6166494c54feb92b0137265d657a0949/substrate/client/cli/src/lib.rs#L243-L245

Took some inspiration from:
https://github.com/paritytech/polkadot-sdk/blob/dc5f96dc6166494c54feb92b0137265d657a0949/substrate/client/cli/src/config.rs#L239-L245

Will look into initializing a temp logger, or remove the logging hook which uses this config cli 🤔

Reply:
Found an easier solution to this, print the warning after logging initialization.
After we do a release with this included, then we'll have to return in `network_config` an error and left a todo to not forget about it